seungki/app.js

```diff
@@ -0,0 +1,201 @@
+const http = require('http');
+
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+const { DataSource } = require('typeorm');
+
+const app = express();
+app.use(express.json());
+app.use(cors());
+app.use(morgan('dev'));
+
+const mysqlDatabase = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE,
+});
+
+mysqlDatabase
+    .initialize()
+    .then(() => {
+        console.log('Data Source has been inilialized!💡');
+    })
+    .catch((err) => {
+        console.error('Error during Data Source initialization', err);
+        mysqlDatabase.destroy();
+    });
+
+// Health Check
+app.get('/ping', (req, res) => {
+    res.status(200).json({ message: 'pongsssss' });
+});
+
+// 유저 회원가입
+app.post('/email_signup', async (req, res) => {
+    const { name, email, profileImage, password } = req.body;
+
+    await mysqlDatabase.query(
+        `INSERT INTO users(
+      name,
+      email,
+      profile_image,
+      password
+    ) VALUES (?, ?, ?, ?);
+    `,
+        [name, email, profileImage, password]
+    );
+    res.status(201).json({ message: 'userCreated' });
+});
+
+// 게시물 등록하기
+app.post('/post_created', async (req, res) => {
+    const { title, content, postImage, userId } = req.body;
+
+    await mysqlDatabase.query(
+        `INSERT INTO posts(
+      title,
+      content,
+      post_image,
+      user_id
+    ) VALUES (?, ?, ?, ?);
+    `,
+        [title, content, postImage, userId]
+    );
+    res.status(201).json({ message: 'postCreated' });
```

Comment:
아래의 설명을 보고 쿼리문을 수정해주세요!
- 가독성 및 유지 보수를 위해 최대한 한눈에 보기 쉽게 작성하는 것이 좋습니다.
- Prettier와 같은 포맷터를 쓰더라도 String은 정렬해주지 않기 때문에 직접 수정해야 합니다.
```suggestion
const queryResult = await AppDataSource.query(
  `
    SELECT
      u.id,
      u.name
    FROM users u
    WHERE u.id = ?
  `, [userId]
)
```